src/client/Client.cc

```diff
@@ -6649,11 +6649,16 @@ void Client::_unmount(bool abort)
   }
 
   mount_cond.wait(lock, [this] {
-    if (!mds_requests.empty()) {
-      ldout(cct, 10) << "waiting on " << mds_requests.size() << " requests"
-		     << dendl;
+    // Only wait for write OPs
+    for (auto& [tid, req] : mds_requests) {
+      if (req->is_write()) {
+        ldout(cct, 10) << "waiting for write requests to complete, "
```

Comment:
Logging here along with the request tid would be better.

Reply:
Make sense to me. Thanks @neesingh-rh